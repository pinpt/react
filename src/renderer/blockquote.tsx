import { registerNode, recurseIntoChildren, NodeProps } from './register';

const BlockQuote = ({ node }: NodeProps) => {
	return <blockquote>{recurseIntoChildren(node)}</blockquote>;
};

registerNode('blockquote', (node) => <BlockQuote node={node} />);
