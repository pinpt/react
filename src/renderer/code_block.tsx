import { Prism as SyntaxHighlighter } from 'react-syntax-highlighter';
import { twilight } from 'react-syntax-highlighter/dist/cjs/styles/prism';
import { NodeProps, registerNode } from './register';

const CodeBlock = ({ node }: NodeProps) => {
	return (
		<SyntaxHighlighter
			language={node.attrs.language}
			style={twilight}
			customStyle={{
				background: 'black',
				fontFamily: 'ui-monospace',
				borderWidth: '0px',
				borderRadius: '.25em',
			}}
		>
			{node.content?.[0].text ?? ''}
		</SyntaxHighlighter>
	);
};

registerNode('code_block', (node) => <CodeBlock node={node} />);
