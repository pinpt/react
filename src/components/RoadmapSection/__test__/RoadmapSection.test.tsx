import renderer from 'react-test-renderer';
import RoadmapCard from '../../RoadmapCard';
import RoadmapSection from '../index';

test('Test default', () => {
	const component = renderer.create(
		<RoadmapSection title="Section Title" description="Some information about what this particular section contains.">
			<RoadmapCard title="My cool feature" description="Some interesting points about the cool feature I made.">
				<div>Point number 1 for the feature.</div>
				<div>Another point for the feature.</div>
				<div>The third and final thing about the cool feature.</div>
			</RoadmapCard>
			<RoadmapCard title="My cool feature" description="Some interesting points about the cool feature I made.">
				<div>Point number 1 for the feature.</div>
				<div>Another point for the feature.</div>
				<div>The third and final thing about the cool feature.</div>
			</RoadmapCard>
			<RoadmapCard title="My cool feature" description="Some interesting points about the cool feature I made.">
				<div>Point number 1 for the feature.</div>
				<div>Another point for the feature.</div>
				<div>The third and final thing about the cool feature.</div>
			</RoadmapCard>
		</RoadmapSection>
	);
	const tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});

test('Test no description', () => {
	const component = renderer.create(
		<RoadmapSection title="Section Title">
			<RoadmapCard title="My cool feature" description="Some interesting points about the cool feature I made.">
				<div>Point number 1 for the feature.</div>
				<div>Another point for the feature.</div>
				<div>The third and final thing about the cool feature.</div>
			</RoadmapCard>
			<RoadmapCard title="My cool feature" description="Some interesting points about the cool feature I made.">
				<div>Point number 1 for the feature.</div>
				<div>Another point for the feature.</div>
				<div>The third and final thing about the cool feature.</div>
			</RoadmapCard>
			<RoadmapCard title="My cool feature" description="Some interesting points about the cool feature I made.">
				<div>Point number 1 for the feature.</div>
				<div>Another point for the feature.</div>
				<div>The third and final thing about the cool feature.</div>
			</RoadmapCard>
		</RoadmapSection>
	);
	const tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});

test('Test default closed', () => {
	const component = renderer.create(
		<RoadmapSection
			title="Section Title"
			description="Some information about what this particular section contains."
			initialOpen={false}
		>
			<RoadmapCard title="My cool feature" description="Some interesting points about the cool feature I made.">
				<div>Point number 1 for the feature.</div>
				<div>Another point for the feature.</div>
				<div>The third and final thing about the cool feature.</div>
			</RoadmapCard>
			<RoadmapCard title="My cool feature" description="Some interesting points about the cool feature I made.">
				<div>Point number 1 for the feature.</div>
				<div>Another point for the feature.</div>
				<div>The third and final thing about the cool feature.</div>
			</RoadmapCard>
			<RoadmapCard title="My cool feature" description="Some interesting points about the cool feature I made.">
				<div>Point number 1 for the feature.</div>
				<div>Another point for the feature.</div>
				<div>The third and final thing about the cool feature.</div>
			</RoadmapCard>
		</RoadmapSection>
	);
	const tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});

test('Test empty section', () => {
	const component = renderer.create(
		<RoadmapSection
			title="Section Title"
			description="Some information about what this particular section contains."
			initialOpen={true}
		/>
	);
	const tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});

test('Test custom className', () => {
	const component = renderer.create(
		<RoadmapSection
			className="test-custom"
			title="Section Title"
			description="Some information about what this particular section contains."
		>
			<RoadmapCard title="My cool feature" description="Some interesting points about the cool feature I made.">
				<div>Point number 1 for the feature.</div>
				<div>Another point for the feature.</div>
				<div>The third and final thing about the cool feature.</div>
			</RoadmapCard>
			<RoadmapCard title="My cool feature" description="Some interesting points about the cool feature I made.">
				<div>Point number 1 for the feature.</div>
				<div>Another point for the feature.</div>
				<div>The third and final thing about the cool feature.</div>
			</RoadmapCard>
			<RoadmapCard title="My cool feature" description="Some interesting points about the cool feature I made.">
				<div>Point number 1 for the feature.</div>
				<div>Another point for the feature.</div>
				<div>The third and final thing about the cool feature.</div>
			</RoadmapCard>
		</RoadmapSection>
	);
	const tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});
