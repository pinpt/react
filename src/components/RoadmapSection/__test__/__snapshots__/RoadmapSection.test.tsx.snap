// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test custom className 1`] = `
<div
  className="Pinpoint RoadmapSection Wrapper test-custom"
>
  <div
    className="Pinpoint RoadmapSection Title"
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-chevron-down fa-w-14 fa-fw Pinpoint RoadmapSection Icon"
      data-icon="chevron-down"
      data-prefix="fas"
      focusable="false"
      onClick={[Function]}
      role="img"
      style={Object {}}
      viewBox="0 0 448 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M207.029 381.476L12.686 187.132c-9.373-9.373-9.373-24.569 0-33.941l22.667-22.667c9.357-9.357 24.522-9.375 33.901-.04L224 284.505l154.745-154.021c9.379-9.335 24.544-9.317 33.901.04l22.667 22.667c9.373 9.373 9.373 24.569 0 33.941L240.971 381.476c-9.373 9.372-24.569 9.372-33.942 0z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
    Section Title
  </div>
  <div
    className="Pinpoint RoadmapSection Description"
  >
    Some information about what this particular section contains.
  </div>
  <div
    className="Pinpoint RoadmapSection Children"
  >
    <div
      className="Pinpoint RoadmapCard Wrapper "
    >
      <div
        className="Pinpoint RoadmapCard Heading"
      >
        <div
          className="Pinpoint RoadmapCard Top"
        >
          <div
            className="Pinpoint RoadmapCard Title"
          >
            My cool feature
          </div>
        </div>
        <div
          className="Pinpoint RoadmapCard Description"
        >
          Some interesting points about the cool feature I made.
        </div>
      </div>
      <div
        className="Pinpoint RoadmapCard Body"
      >
        <div>
          Point number 1 for the feature.
        </div>
        <div>
          Another point for the feature.
        </div>
        <div>
          The third and final thing about the cool feature.
        </div>
      </div>
      <div
        className="Pinpoint RoadmapCard Actions"
      >
        <div
          className="Pinpoint RoadmapCard VoteButton"
        >
          <div
            className="Pinpoint Vote Wrapper  "
          >
            <div
              className="Pinpoint Vote Icon"
              onClick={[Function]}
            >
              <svg
                aria-hidden="true"
                className="svg-inline--fa fa-vote-yea fa-w-20 fa-fw Pinpoint Vote Button "
                data-icon="vote-yea"
                data-prefix="fas"
                focusable="false"
                role="img"
                style={Object {}}
                viewBox="0 0 640 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M608 320h-64v64h22.4c5.3 0 9.6 3.6 9.6 8v16c0 4.4-4.3 8-9.6 8H73.6c-5.3 0-9.6-3.6-9.6-8v-16c0-4.4 4.3-8 9.6-8H96v-64H32c-17.7 0-32 14.3-32 32v96c0 17.7 14.3 32 32 32h576c17.7 0 32-14.3 32-32v-96c0-17.7-14.3-32-32-32zm-96 64V64.3c0-17.9-14.5-32.3-32.3-32.3H160.4C142.5 32 128 46.5 128 64.3V384h384zM211.2 202l25.5-25.3c4.2-4.2 11-4.2 15.2.1l41.3 41.6 95.2-94.4c4.2-4.2 11-4.2 15.2.1l25.3 25.5c4.2 4.2 4.2 11-.1 15.2L300.5 292c-4.2 4.2-11 4.2-15.2-.1l-74.1-74.7c-4.3-4.2-4.2-11 0-15.2z"
                  fill="currentColor"
                  style={Object {}}
                />
              </svg>
            </div>
          </div>
        </div>
        <div
          className="Pinpoint RoadmapCard Count"
        >
          0
           vote
          s
        </div>
      </div>
    </div>
    <div
      className="Pinpoint RoadmapCard Wrapper "
    >
      <div
        className="Pinpoint RoadmapCard Heading"
      >
        <div
          className="Pinpoint RoadmapCard Top"
        >
          <div
            className="Pinpoint RoadmapCard Title"
          >
            My cool feature
          </div>
        </div>
        <div
          className="Pinpoint RoadmapCard Description"
        >
          Some interesting points about the cool feature I made.
        </div>
      </div>
      <div
        className="Pinpoint RoadmapCard Body"
      >
        <div>
          Point number 1 for the feature.
        </div>
        <div>
          Another point for the feature.
        </div>
        <div>
          The third and final thing about the cool feature.
        </div>
      </div>
      <div
        className="Pinpoint RoadmapCard Actions"
      >
        <div
          className="Pinpoint RoadmapCard VoteButton"
        >
          <div
            className="Pinpoint Vote Wrapper  "
          >
            <div
              className="Pinpoint Vote Icon"
              onClick={[Function]}
            >
              <svg
                aria-hidden="true"
                className="svg-inline--fa fa-vote-yea fa-w-20 fa-fw Pinpoint Vote Button "
                data-icon="vote-yea"
                data-prefix="fas"
                focusable="false"
                role="img"
                style={Object {}}
                viewBox="0 0 640 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M608 320h-64v64h22.4c5.3 0 9.6 3.6 9.6 8v16c0 4.4-4.3 8-9.6 8H73.6c-5.3 0-9.6-3.6-9.6-8v-16c0-4.4 4.3-8 9.6-8H96v-64H32c-17.7 0-32 14.3-32 32v96c0 17.7 14.3 32 32 32h576c17.7 0 32-14.3 32-32v-96c0-17.7-14.3-32-32-32zm-96 64V64.3c0-17.9-14.5-32.3-32.3-32.3H160.4C142.5 32 128 46.5 128 64.3V384h384zM211.2 202l25.5-25.3c4.2-4.2 11-4.2 15.2.1l41.3 41.6 95.2-94.4c4.2-4.2 11-4.2 15.2.1l25.3 25.5c4.2 4.2 4.2 11-.1 15.2L300.5 292c-4.2 4.2-11 4.2-15.2-.1l-74.1-74.7c-4.3-4.2-4.2-11 0-15.2z"
                  fill="currentColor"
                  style={Object {}}
                />
              </svg>
            </div>
          </div>
        </div>
        <div
          className="Pinpoint RoadmapCard Count"
        >
          0
           vote
          s
        </div>
      </div>
    </div>
    <div
      className="Pinpoint RoadmapCard Wrapper "
    >
      <div
        className="Pinpoint RoadmapCard Heading"
      >
        <div
          className="Pinpoint RoadmapCard Top"
        >
          <div
            className="Pinpoint RoadmapCard Title"
          >
            My cool feature
          </div>
        </div>
        <div
          className="Pinpoint RoadmapCard Description"
        >
          Some interesting points about the cool feature I made.
        </div>
      </div>
      <div
        className="Pinpoint RoadmapCard Body"
      >
        <div>
          Point number 1 for the feature.
        </div>
        <div>
          Another point for the feature.
        </div>
        <div>
          The third and final thing about the cool feature.
        </div>
      </div>
      <div
        className="Pinpoint RoadmapCard Actions"
      >
        <div
          className="Pinpoint RoadmapCard VoteButton"
        >
          <div
            className="Pinpoint Vote Wrapper  "
          >
            <div
              className="Pinpoint Vote Icon"
              onClick={[Function]}
            >
              <svg
                aria-hidden="true"
                className="svg-inline--fa fa-vote-yea fa-w-20 fa-fw Pinpoint Vote Button "
                data-icon="vote-yea"
                data-prefix="fas"
                focusable="false"
                role="img"
                style={Object {}}
                viewBox="0 0 640 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M608 320h-64v64h22.4c5.3 0 9.6 3.6 9.6 8v16c0 4.4-4.3 8-9.6 8H73.6c-5.3 0-9.6-3.6-9.6-8v-16c0-4.4 4.3-8 9.6-8H96v-64H32c-17.7 0-32 14.3-32 32v96c0 17.7 14.3 32 32 32h576c17.7 0 32-14.3 32-32v-96c0-17.7-14.3-32-32-32zm-96 64V64.3c0-17.9-14.5-32.3-32.3-32.3H160.4C142.5 32 128 46.5 128 64.3V384h384zM211.2 202l25.5-25.3c4.2-4.2 11-4.2 15.2.1l41.3 41.6 95.2-94.4c4.2-4.2 11-4.2 15.2.1l25.3 25.5c4.2 4.2 4.2 11-.1 15.2L300.5 292c-4.2 4.2-11 4.2-15.2-.1l-74.1-74.7c-4.3-4.2-4.2-11 0-15.2z"
                  fill="currentColor"
                  style={Object {}}
                />
              </svg>
            </div>
          </div>
        </div>
        <div
          className="Pinpoint RoadmapCard Count"
        >
          0
           vote
          s
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Test default 1`] = `
<div
  className="Pinpoint RoadmapSection Wrapper "
>
  <div
    className="Pinpoint RoadmapSection Title"
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-chevron-down fa-w-14 fa-fw Pinpoint RoadmapSection Icon"
      data-icon="chevron-down"
      data-prefix="fas"
      focusable="false"
      onClick={[Function]}
      role="img"
      style={Object {}}
      viewBox="0 0 448 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M207.029 381.476L12.686 187.132c-9.373-9.373-9.373-24.569 0-33.941l22.667-22.667c9.357-9.357 24.522-9.375 33.901-.04L224 284.505l154.745-154.021c9.379-9.335 24.544-9.317 33.901.04l22.667 22.667c9.373 9.373 9.373 24.569 0 33.941L240.971 381.476c-9.373 9.372-24.569 9.372-33.942 0z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
    Section Title
  </div>
  <div
    className="Pinpoint RoadmapSection Description"
  >
    Some information about what this particular section contains.
  </div>
  <div
    className="Pinpoint RoadmapSection Children"
  >
    <div
      className="Pinpoint RoadmapCard Wrapper "
    >
      <div
        className="Pinpoint RoadmapCard Heading"
      >
        <div
          className="Pinpoint RoadmapCard Top"
        >
          <div
            className="Pinpoint RoadmapCard Title"
          >
            My cool feature
          </div>
        </div>
        <div
          className="Pinpoint RoadmapCard Description"
        >
          Some interesting points about the cool feature I made.
        </div>
      </div>
      <div
        className="Pinpoint RoadmapCard Body"
      >
        <div>
          Point number 1 for the feature.
        </div>
        <div>
          Another point for the feature.
        </div>
        <div>
          The third and final thing about the cool feature.
        </div>
      </div>
      <div
        className="Pinpoint RoadmapCard Actions"
      >
        <div
          className="Pinpoint RoadmapCard VoteButton"
        >
          <div
            className="Pinpoint Vote Wrapper  "
          >
            <div
              className="Pinpoint Vote Icon"
              onClick={[Function]}
            >
              <svg
                aria-hidden="true"
                className="svg-inline--fa fa-vote-yea fa-w-20 fa-fw Pinpoint Vote Button "
                data-icon="vote-yea"
                data-prefix="fas"
                focusable="false"
                role="img"
                style={Object {}}
                viewBox="0 0 640 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M608 320h-64v64h22.4c5.3 0 9.6 3.6 9.6 8v16c0 4.4-4.3 8-9.6 8H73.6c-5.3 0-9.6-3.6-9.6-8v-16c0-4.4 4.3-8 9.6-8H96v-64H32c-17.7 0-32 14.3-32 32v96c0 17.7 14.3 32 32 32h576c17.7 0 32-14.3 32-32v-96c0-17.7-14.3-32-32-32zm-96 64V64.3c0-17.9-14.5-32.3-32.3-32.3H160.4C142.5 32 128 46.5 128 64.3V384h384zM211.2 202l25.5-25.3c4.2-4.2 11-4.2 15.2.1l41.3 41.6 95.2-94.4c4.2-4.2 11-4.2 15.2.1l25.3 25.5c4.2 4.2 4.2 11-.1 15.2L300.5 292c-4.2 4.2-11 4.2-15.2-.1l-74.1-74.7c-4.3-4.2-4.2-11 0-15.2z"
                  fill="currentColor"
                  style={Object {}}
                />
              </svg>
            </div>
          </div>
        </div>
        <div
          className="Pinpoint RoadmapCard Count"
        >
          0
           vote
          s
        </div>
      </div>
    </div>
    <div
      className="Pinpoint RoadmapCard Wrapper "
    >
      <div
        className="Pinpoint RoadmapCard Heading"
      >
        <div
          className="Pinpoint RoadmapCard Top"
        >
          <div
            className="Pinpoint RoadmapCard Title"
          >
            My cool feature
          </div>
        </div>
        <div
          className="Pinpoint RoadmapCard Description"
        >
          Some interesting points about the cool feature I made.
        </div>
      </div>
      <div
        className="Pinpoint RoadmapCard Body"
      >
        <div>
          Point number 1 for the feature.
        </div>
        <div>
          Another point for the feature.
        </div>
        <div>
          The third and final thing about the cool feature.
        </div>
      </div>
      <div
        className="Pinpoint RoadmapCard Actions"
      >
        <div
          className="Pinpoint RoadmapCard VoteButton"
        >
          <div
            className="Pinpoint Vote Wrapper  "
          >
            <div
              className="Pinpoint Vote Icon"
              onClick={[Function]}
            >
              <svg
                aria-hidden="true"
                className="svg-inline--fa fa-vote-yea fa-w-20 fa-fw Pinpoint Vote Button "
                data-icon="vote-yea"
                data-prefix="fas"
                focusable="false"
                role="img"
                style={Object {}}
                viewBox="0 0 640 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M608 320h-64v64h22.4c5.3 0 9.6 3.6 9.6 8v16c0 4.4-4.3 8-9.6 8H73.6c-5.3 0-9.6-3.6-9.6-8v-16c0-4.4 4.3-8 9.6-8H96v-64H32c-17.7 0-32 14.3-32 32v96c0 17.7 14.3 32 32 32h576c17.7 0 32-14.3 32-32v-96c0-17.7-14.3-32-32-32zm-96 64V64.3c0-17.9-14.5-32.3-32.3-32.3H160.4C142.5 32 128 46.5 128 64.3V384h384zM211.2 202l25.5-25.3c4.2-4.2 11-4.2 15.2.1l41.3 41.6 95.2-94.4c4.2-4.2 11-4.2 15.2.1l25.3 25.5c4.2 4.2 4.2 11-.1 15.2L300.5 292c-4.2 4.2-11 4.2-15.2-.1l-74.1-74.7c-4.3-4.2-4.2-11 0-15.2z"
                  fill="currentColor"
                  style={Object {}}
                />
              </svg>
            </div>
          </div>
        </div>
        <div
          className="Pinpoint RoadmapCard Count"
        >
          0
           vote
          s
        </div>
      </div>
    </div>
    <div
      className="Pinpoint RoadmapCard Wrapper "
    >
      <div
        className="Pinpoint RoadmapCard Heading"
      >
        <div
          className="Pinpoint RoadmapCard Top"
        >
          <div
            className="Pinpoint RoadmapCard Title"
          >
            My cool feature
          </div>
        </div>
        <div
          className="Pinpoint RoadmapCard Description"
        >
          Some interesting points about the cool feature I made.
        </div>
      </div>
      <div
        className="Pinpoint RoadmapCard Body"
      >
        <div>
          Point number 1 for the feature.
        </div>
        <div>
          Another point for the feature.
        </div>
        <div>
          The third and final thing about the cool feature.
        </div>
      </div>
      <div
        className="Pinpoint RoadmapCard Actions"
      >
        <div
          className="Pinpoint RoadmapCard VoteButton"
        >
          <div
            className="Pinpoint Vote Wrapper  "
          >
            <div
              className="Pinpoint Vote Icon"
              onClick={[Function]}
            >
              <svg
                aria-hidden="true"
                className="svg-inline--fa fa-vote-yea fa-w-20 fa-fw Pinpoint Vote Button "
                data-icon="vote-yea"
                data-prefix="fas"
                focusable="false"
                role="img"
                style={Object {}}
                viewBox="0 0 640 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M608 320h-64v64h22.4c5.3 0 9.6 3.6 9.6 8v16c0 4.4-4.3 8-9.6 8H73.6c-5.3 0-9.6-3.6-9.6-8v-16c0-4.4 4.3-8 9.6-8H96v-64H32c-17.7 0-32 14.3-32 32v96c0 17.7 14.3 32 32 32h576c17.7 0 32-14.3 32-32v-96c0-17.7-14.3-32-32-32zm-96 64V64.3c0-17.9-14.5-32.3-32.3-32.3H160.4C142.5 32 128 46.5 128 64.3V384h384zM211.2 202l25.5-25.3c4.2-4.2 11-4.2 15.2.1l41.3 41.6 95.2-94.4c4.2-4.2 11-4.2 15.2.1l25.3 25.5c4.2 4.2 4.2 11-.1 15.2L300.5 292c-4.2 4.2-11 4.2-15.2-.1l-74.1-74.7c-4.3-4.2-4.2-11 0-15.2z"
                  fill="currentColor"
                  style={Object {}}
                />
              </svg>
            </div>
          </div>
        </div>
        <div
          className="Pinpoint RoadmapCard Count"
        >
          0
           vote
          s
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Test default closed 1`] = `
<div
  className="Pinpoint RoadmapSection Wrapper "
>
  <div
    className="Pinpoint RoadmapSection Title"
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-chevron-right fa-w-10 fa-fw Pinpoint RoadmapSection Icon"
      data-icon="chevron-right"
      data-prefix="fas"
      focusable="false"
      onClick={[Function]}
      role="img"
      style={Object {}}
      viewBox="0 0 320 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M285.476 272.971L91.132 467.314c-9.373 9.373-24.569 9.373-33.941 0l-22.667-22.667c-9.357-9.357-9.375-24.522-.04-33.901L188.505 256 34.484 101.255c-9.335-9.379-9.317-24.544.04-33.901l22.667-22.667c9.373-9.373 24.569-9.373 33.941 0L285.475 239.03c9.373 9.372 9.373 24.568.001 33.941z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
    Section Title
  </div>
  <div
    className="Pinpoint RoadmapSection Description"
  >
    Some information about what this particular section contains.
  </div>
</div>
`;

exports[`Test empty section 1`] = `
<div
  className="Pinpoint RoadmapSection Wrapper "
>
  <div
    className="Pinpoint RoadmapSection Title"
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-chevron-down fa-w-14 fa-fw Pinpoint RoadmapSection Icon"
      data-icon="chevron-down"
      data-prefix="fas"
      focusable="false"
      onClick={[Function]}
      role="img"
      style={Object {}}
      viewBox="0 0 448 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M207.029 381.476L12.686 187.132c-9.373-9.373-9.373-24.569 0-33.941l22.667-22.667c9.357-9.357 24.522-9.375 33.901-.04L224 284.505l154.745-154.021c9.379-9.335 24.544-9.317 33.901.04l22.667 22.667c9.373 9.373 9.373 24.569 0 33.941L240.971 381.476c-9.373 9.372-24.569 9.372-33.942 0z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
    Section Title
  </div>
  <div
    className="Pinpoint RoadmapSection Description"
  >
    Some information about what this particular section contains.
  </div>
  <div
    className="Pinpoint RoadmapSection Children"
  >
    <div
      className="Pinpoint RoadmapSection Empty"
    >
      There are currently no items in this stage on the roadmap.
    </div>
  </div>
</div>
`;

exports[`Test no description 1`] = `
<div
  className="Pinpoint RoadmapSection Wrapper "
>
  <div
    className="Pinpoint RoadmapSection Title"
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-chevron-down fa-w-14 fa-fw Pinpoint RoadmapSection Icon"
      data-icon="chevron-down"
      data-prefix="fas"
      focusable="false"
      onClick={[Function]}
      role="img"
      style={Object {}}
      viewBox="0 0 448 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M207.029 381.476L12.686 187.132c-9.373-9.373-9.373-24.569 0-33.941l22.667-22.667c9.357-9.357 24.522-9.375 33.901-.04L224 284.505l154.745-154.021c9.379-9.335 24.544-9.317 33.901.04l22.667 22.667c9.373 9.373 9.373 24.569 0 33.941L240.971 381.476c-9.373 9.372-24.569 9.372-33.942 0z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
    Section Title
  </div>
  <div
    className="Pinpoint RoadmapSection Children"
  >
    <div
      className="Pinpoint RoadmapCard Wrapper "
    >
      <div
        className="Pinpoint RoadmapCard Heading"
      >
        <div
          className="Pinpoint RoadmapCard Top"
        >
          <div
            className="Pinpoint RoadmapCard Title"
          >
            My cool feature
          </div>
        </div>
        <div
          className="Pinpoint RoadmapCard Description"
        >
          Some interesting points about the cool feature I made.
        </div>
      </div>
      <div
        className="Pinpoint RoadmapCard Body"
      >
        <div>
          Point number 1 for the feature.
        </div>
        <div>
          Another point for the feature.
        </div>
        <div>
          The third and final thing about the cool feature.
        </div>
      </div>
      <div
        className="Pinpoint RoadmapCard Actions"
      >
        <div
          className="Pinpoint RoadmapCard VoteButton"
        >
          <div
            className="Pinpoint Vote Wrapper  "
          >
            <div
              className="Pinpoint Vote Icon"
              onClick={[Function]}
            >
              <svg
                aria-hidden="true"
                className="svg-inline--fa fa-vote-yea fa-w-20 fa-fw Pinpoint Vote Button "
                data-icon="vote-yea"
                data-prefix="fas"
                focusable="false"
                role="img"
                style={Object {}}
                viewBox="0 0 640 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M608 320h-64v64h22.4c5.3 0 9.6 3.6 9.6 8v16c0 4.4-4.3 8-9.6 8H73.6c-5.3 0-9.6-3.6-9.6-8v-16c0-4.4 4.3-8 9.6-8H96v-64H32c-17.7 0-32 14.3-32 32v96c0 17.7 14.3 32 32 32h576c17.7 0 32-14.3 32-32v-96c0-17.7-14.3-32-32-32zm-96 64V64.3c0-17.9-14.5-32.3-32.3-32.3H160.4C142.5 32 128 46.5 128 64.3V384h384zM211.2 202l25.5-25.3c4.2-4.2 11-4.2 15.2.1l41.3 41.6 95.2-94.4c4.2-4.2 11-4.2 15.2.1l25.3 25.5c4.2 4.2 4.2 11-.1 15.2L300.5 292c-4.2 4.2-11 4.2-15.2-.1l-74.1-74.7c-4.3-4.2-4.2-11 0-15.2z"
                  fill="currentColor"
                  style={Object {}}
                />
              </svg>
            </div>
          </div>
        </div>
        <div
          className="Pinpoint RoadmapCard Count"
        >
          0
           vote
          s
        </div>
      </div>
    </div>
    <div
      className="Pinpoint RoadmapCard Wrapper "
    >
      <div
        className="Pinpoint RoadmapCard Heading"
      >
        <div
          className="Pinpoint RoadmapCard Top"
        >
          <div
            className="Pinpoint RoadmapCard Title"
          >
            My cool feature
          </div>
        </div>
        <div
          className="Pinpoint RoadmapCard Description"
        >
          Some interesting points about the cool feature I made.
        </div>
      </div>
      <div
        className="Pinpoint RoadmapCard Body"
      >
        <div>
          Point number 1 for the feature.
        </div>
        <div>
          Another point for the feature.
        </div>
        <div>
          The third and final thing about the cool feature.
        </div>
      </div>
      <div
        className="Pinpoint RoadmapCard Actions"
      >
        <div
          className="Pinpoint RoadmapCard VoteButton"
        >
          <div
            className="Pinpoint Vote Wrapper  "
          >
            <div
              className="Pinpoint Vote Icon"
              onClick={[Function]}
            >
              <svg
                aria-hidden="true"
                className="svg-inline--fa fa-vote-yea fa-w-20 fa-fw Pinpoint Vote Button "
                data-icon="vote-yea"
                data-prefix="fas"
                focusable="false"
                role="img"
                style={Object {}}
                viewBox="0 0 640 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M608 320h-64v64h22.4c5.3 0 9.6 3.6 9.6 8v16c0 4.4-4.3 8-9.6 8H73.6c-5.3 0-9.6-3.6-9.6-8v-16c0-4.4 4.3-8 9.6-8H96v-64H32c-17.7 0-32 14.3-32 32v96c0 17.7 14.3 32 32 32h576c17.7 0 32-14.3 32-32v-96c0-17.7-14.3-32-32-32zm-96 64V64.3c0-17.9-14.5-32.3-32.3-32.3H160.4C142.5 32 128 46.5 128 64.3V384h384zM211.2 202l25.5-25.3c4.2-4.2 11-4.2 15.2.1l41.3 41.6 95.2-94.4c4.2-4.2 11-4.2 15.2.1l25.3 25.5c4.2 4.2 4.2 11-.1 15.2L300.5 292c-4.2 4.2-11 4.2-15.2-.1l-74.1-74.7c-4.3-4.2-4.2-11 0-15.2z"
                  fill="currentColor"
                  style={Object {}}
                />
              </svg>
            </div>
          </div>
        </div>
        <div
          className="Pinpoint RoadmapCard Count"
        >
          0
           vote
          s
        </div>
      </div>
    </div>
    <div
      className="Pinpoint RoadmapCard Wrapper "
    >
      <div
        className="Pinpoint RoadmapCard Heading"
      >
        <div
          className="Pinpoint RoadmapCard Top"
        >
          <div
            className="Pinpoint RoadmapCard Title"
          >
            My cool feature
          </div>
        </div>
        <div
          className="Pinpoint RoadmapCard Description"
        >
          Some interesting points about the cool feature I made.
        </div>
      </div>
      <div
        className="Pinpoint RoadmapCard Body"
      >
        <div>
          Point number 1 for the feature.
        </div>
        <div>
          Another point for the feature.
        </div>
        <div>
          The third and final thing about the cool feature.
        </div>
      </div>
      <div
        className="Pinpoint RoadmapCard Actions"
      >
        <div
          className="Pinpoint RoadmapCard VoteButton"
        >
          <div
            className="Pinpoint Vote Wrapper  "
          >
            <div
              className="Pinpoint Vote Icon"
              onClick={[Function]}
            >
              <svg
                aria-hidden="true"
                className="svg-inline--fa fa-vote-yea fa-w-20 fa-fw Pinpoint Vote Button "
                data-icon="vote-yea"
                data-prefix="fas"
                focusable="false"
                role="img"
                style={Object {}}
                viewBox="0 0 640 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M608 320h-64v64h22.4c5.3 0 9.6 3.6 9.6 8v16c0 4.4-4.3 8-9.6 8H73.6c-5.3 0-9.6-3.6-9.6-8v-16c0-4.4 4.3-8 9.6-8H96v-64H32c-17.7 0-32 14.3-32 32v96c0 17.7 14.3 32 32 32h576c17.7 0 32-14.3 32-32v-96c0-17.7-14.3-32-32-32zm-96 64V64.3c0-17.9-14.5-32.3-32.3-32.3H160.4C142.5 32 128 46.5 128 64.3V384h384zM211.2 202l25.5-25.3c4.2-4.2 11-4.2 15.2.1l41.3 41.6 95.2-94.4c4.2-4.2 11-4.2 15.2.1l25.3 25.5c4.2 4.2 4.2 11-.1 15.2L300.5 292c-4.2 4.2-11 4.2-15.2-.1l-74.1-74.7c-4.3-4.2-4.2-11 0-15.2z"
                  fill="currentColor"
                  style={Object {}}
                />
              </svg>
            </div>
          </div>
        </div>
        <div
          className="Pinpoint RoadmapCard Count"
        >
          0
           vote
          s
        </div>
      </div>
    </div>
  </div>
</div>
`;
